### PR TITLE
backport for add_xcom_sidecar

### LINKED ISSUE #11298: Kubernetes - related backports needs some fixes for xcom cases
The "cncf.kubernetes" still has problems when releasing backport packages.

Reported by @mik-laj in https://lists.apache.org/thread.html/rd7b8fa87ac8117812e9883b342876eaee6039ce8e421cab3e454406c%40%3Cdev.airflow.apache.org%3E:

**Apache Airflow version**:

Master Backport packages for "cncf.kubernetes" and "google" 
Installed on top of 1.10.12

**What happened**:

The example_spark_kubernetes_spark_pi.yaml from our repository and example DAG works

  - airflow/providers/cncf/kubernetes/example_dags/example_kubernetes.py, 
  - airflow/providers/cncf/kubernetes/example_dags/example_spark_kubernetes_spark_pi.yaml

But it does not work with `example_kubernetes_operator`. I performed the following steps which ended with an error.

```
$ curl -LO https://raw.githubusercontent.com/apache/airflow/master/airflow/providers/cncf/kubernetes/example_dags/example_kubernetes.py
$ cat example_kubernetes.py | sed 's/operators.bash/operators.bash_operator/g' | sed 's/in_cluster=True/in_cluster=False/g' > a.py
$ mv a.py example_kubernetes.py
$ airflow test example_kubernetes_operator write-xcom 2010-01-01 
.....
[2020-10-04 23:29:06,623] {taskinstance.py:1150} ERROR - type object 'PodGenerator' has no attribute 'add_xcom_sidecar'
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 984, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py", line 269, in execute
    self.pod = self.create_pod_request_obj()
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py", line 405, in create_pod_request_obj
    pod = PodGenerator.add_xcom_sidecar(pod)
AttributeError: type object 'PodGenerator' has no attribute 'add_xcom_sidecar'
```

### PR DESCRIPTION
**Pull request by dimberman**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by dimberman** [comment]
cc: @michalmisiewicz

**Comment by mik-laj** [comment]
@michalslowikowski00 Can you look at it?

File: provider_packages/refactor_provider_packages.py [hunk 1]
```diff
@@ -650,6 +650,14 @@ def odbc_package_filter(node: LN, capture: Capture, filename: Filename) -> bool:
...
         )
 
+    def refactor_kubernetes_pod_operator(self):
+        (
+            self.qry.
+            select_class("airflow.kubernetes.pod_generator.PodGenerator").
+            select_method("add_xcom_sidecar").
+            rename("add_sidecar")
+        )
+
```
Comment (kaxil) [inline provider_packages/refactor_provider_packages.py:664]: You will need to add this method in Line 669 too

Comment (dimberman) [reply provider_packages/refactor_provider_packages.py:664]: @kaxil oof thanks :)

**Review by mik-laj — APPROVED** [review]

**Comment by michalslowikowski00** [comment]
Is it related to [11298](https://github.com/apache/airflow/issues/11298)?

**Comment by mik-laj** [comment]
@michalslowikowski00  Yes. Can you test it?

**Review by kaxil — APPROVED** [review]

**Comment by dimberman** [comment]
@michalslowikowski00 if you have a chance to test it I'd appreciate it, can merge once I hear back :)

**Comment by michalslowikowski00** [comment]
Hey @dimberman. 
I just tested the fix and it doesn't work. Sorry. :(
I will try to figure it out.

**Comment by mik-laj** [comment]
Changing `add_xcom_sidecar` to `add_sidecar` works, but the Bowler code does not make expected change. I together with @michalslowikowski00  made these changes manually to test it.

**Comment by dimberman** [comment]
@mik-laj Any idea why bowler isn't making the change?

**Comment by dimberman** [comment]
Maybe I need to remove the filter?

File: provider_packages/refactor_provider_packages.py [hunk 2]
```diff
@@ -650,6 +650,18 @@ def odbc_package_filter(node: LN, capture: Capture, filename: Filename) -> bool:
             rename("airflow.providers.odbc.utils.helpers")
         )
 
+    def refactor_kubernetes_pod_operator(self):
+        def kubernetes_package_filter(node: LN, capture: Capture, filename: Filename) -> bool:
+            return filename.startswith("./airflow/providers/cncf/kubernetes")
+
+        (
+            self.qry.
+            select_class("airflow.kubernetes.pod_generator.PodGenerator").
```
Comment (potiuk) [inline provider_packages/refactor_provider_packages.py]: I believe select_class works only on  name not whole class including package/module. See some examples below. If you want to perform an operation on specific class in specific module, you can provide filename filters I think.

Comment (potiuk) [reply provider_packages/refactor_provider_packages.py]: @dimberman

Comment (potiuk) [reply provider_packages/refactor_provider_packages.py]: BTW. you can easily test Bowler and debug it. You just need to run this as python code and add `--debug`, see below comment in main (but it is a lot slower and it's better to comment out all other filters only leaving the one you are working on.

**Comment by dimberman** [comment]
@potiuk @mik-laj I got it working 

```
(airflow) new-features  (backport-refactor-k8s-pod-operator)  (⎈ |gke_astronomer-cloud-dev-236021_us-east4-a_dev-cluster|airflow-2-test) $
python provider_packages/refactor_provider_packages.py --debug
--- ./airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py
+++ ./airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py
@@ -407,7 +407,7 @@
         for secret in self.secrets:
             pod = secret.attach_to_pod(pod)
         if self.do_xcom_push:
-            pod = PodGenerator.add_xcom_sidecar(pod)
+            pod = PodGenerator.add_sidecar(pod)
         return pod
```

**Comment by mik-laj** [comment]
🎉 Thanks for this PR. It's just a few lines of code, but very valuable. Many users wait until they can use this operator.

**Comment by potiuk** [comment]
:tada: !

**Comment by michalslowikowski00** [comment]
Great! 🚀

**Pull request merged**